MidasCivil_Adapter/CRUD/Read/Loads/AreaDifferentialTemperatureLoad.cs

```diff
@@ -0,0 +1,130 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using BH.oM.Adapters.MidasCivil;
+using BH.Engine.Adapter;
+using BH.oM.Structure.Loads;
+using BH.oM.Structure.Elements;
+using BH.Engine.Reflection;
+
+namespace BH.Adapter.MidasCivil
+{
+    public partial class MidasCivilAdapter
+    {
+        private List<ILoad> ReadAreaDifferentialTemperatureLoads(List<string> ids = null)
+        {
+            List<ILoad> bhomAreaDifferentialTemperatureLoads = new List<ILoad>();
+
+            List<Loadcase> bhomLoadcases = ReadLoadcases();
+            Dictionary<string, Loadcase> loadcaseDictionary = bhomLoadcases.ToDictionary(
+                        x => x.Name);
+
+            string[] loadcaseFolders = Directory.GetDirectories(m_directory + "\\TextFiles");
+
+            int i = 1;
+
+            foreach (string loadcaseFolder in loadcaseFolders)
+            {
+                string loadcase = Path.GetFileName(loadcaseFolder);
+                List<string> areaDifferentialTemperatureLoadText = GetSectionText(loadcase + "\\THERGRAD");
+                List<string> areaUniformTemperatureLoadText = GetSectionText(loadcase + "\\ELTEMPER");
+                List<string> feMeshComparison = new List<string>();
+                List<string> loadedFEMeshes = new List<string>();
+
+                if (areaDifferentialTemperatureLoadText.Count != 0)
+                {
+                    int Index = 0;
+                    if (areaUniformTemperatureLoadText.Count == areaDifferentialTemperatureLoadText.Count)
+                    {
+                        foreach (string areaDifferentialTemperatureLoad in areaDifferentialTemperatureLoadText)
+                        {
+                            List<string> ADTLdelimitted = areaDifferentialTemperatureLoad.Split(',').ToList();
+                            List<string> AUTLdelimitted = areaUniformTemperatureLoadText[Index].Split(',').ToList();
+                            if (ADTLdelimitted[0].Trim() == AUTLdelimitted[0].Trim())
+                            {
+                                loadedFEMeshes.Add(ADTLdelimitted[0].Trim());
+                                ADTLdelimitted.RemoveAt(0);
+                                ADTLdelimitted.Add(AUTLdelimitted[1].Trim());
```

Comment:
I don't think storing the global temperature as an additional `string` in a `List<string>` is a good way to do this.

Instead, you should have a `List<string>` that has the relevant information for the `AreaDifferentialTemperatureLoad`, and a seperate `List<double>` called `globalTemperature` that stores the offset, if the global temperature does not exist it is simply 0. 

This would be much clearer for someone debugging or reading the code in my opinion.